src/traces/funnelarea/attributes.js

```diff
@@ -0,0 +1,85 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var pieAttrs = require('../pie/attributes');
+var sunburstMarker = require('../sunburst/attributes').marker;
+var plotAttrs = require('../../plots/attributes');
+var domainAttrs = require('../../plots/domain').attributes;
+var hovertemplateAttrs = require('../../components/fx/hovertemplate_attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = {
+    labels: pieAttrs.labels,
+    // equivalent of x0 and dx, if label is missing
+    label0: pieAttrs.label0,
+    dlabel: pieAttrs.dlabel,
+    values: pieAttrs.values,
+    marker: sunburstMarker,
+    text: pieAttrs.text,
+    hovertext: pieAttrs.hovertext,
+
+    scalegroup: extendFlat({}, pieAttrs.scalegroup, {
+        description: [
+            'If there are multiple funnelareas that should be sized according to',
+            'their totals, link them by providing a non-empty group id here',
+            'shared by every trace in the same group.'
+        ].join(' ')
+    }),
+
+    textinfo: extendFlat({}, pieAttrs.textinfo, {
+        flags: ['label', 'text', 'value', 'percent']
+    }),
+
+    hoverinfo: extendFlat({}, plotAttrs.hoverinfo, {
+        flags: ['label', 'text', 'value', 'percent', 'name']
+    }),
+
+    hovertemplate: hovertemplateAttrs({}, {
+        keys: ['label', 'color', 'value', 'percent', 'text']
+    }),
+
+    textfont: pieAttrs.textfont,
+    insidetextfont: pieAttrs.insidetextfont,
+    outsidetextfont: pieAttrs.outsidetextfont,
+
+    title: extendFlat({}, pieAttrs.title, {
```

Comment:
I would prefer declaring `title` as:

```js
title: {
  text: pieAttrs.title,
  font: pieAttrs.font,
  position: extendFlat({}, pieAttrs.position, {
    values: ['top left', 'top center', 'top right']
  }),

  editType: 'plot'
}
```

for clarity.

Reply:
Revised in https://github.com/plotly/plotly.js/pull/3876/commits/38b474d3a4c47bf6a761ac9c0f2d5f0367632d35.
Not clear why `pie` uses `editType: 'calc'`?

Reply:
> Not clear why pie uses editType: 'calc'?

Good :eye: `editType: 'plot'` should suffice for pie and funnelarea `title.*` updates.

Reply:
restyle tests for `title.position` are added in https://github.com/plotly/plotly.js/commit/08eecaa28ba5bb759a3ab6dbc14434704ba679f8 (my branch)

the other attributes that @archmoj switched from editType `'calc'` to `'plot'` are properly tested in `pie_test.js`.

Reply:
Done in https://github.com/plotly/plotly.js/pull/3876/commits/dd673a21af3e9510aa39d17cc2ccb0e25dab91f3